### PR TITLE
Initial swap improvements

### PR DESCRIPTION
**Pull request by onyb**
### Summary

* Disabled the Swap button if not connected to mainnet.
* Improved the layout of the Swap screen, notably putting the asset and the corresponding amount side-by-side.
* Fixed a bug in the currency dropdown fields that prevented changing the asset pairs.

### Screenshots

<table>
  <thead>
    <tr>
      <th>Before (<code>debug/swaps</code>)</th>
      <th>Main screen (<code>onyb/swaps</code>)</th>
      <th>Swap screen (<code>onyb/swaps</code>)</th>
    </tr>
  </thead>
  <tr>
    <td>
      
    </td>
    <td>
      
    </td>
    <td>
      Screenshot 2021-05-10 at 16 24 04
    </td>
  </tr>
</table>

### CODE REVIEW

File: ui/app/ducks/metamask/metamask.js [hunk 1]
```diff
@@ -304,64 +304,39 @@ export default function reduceMetamask (state = {}, action) {
...
-              ...newSwapFrom.unapprovedTxs,
-              [newSwapFrom.editingTransactionId]: {
-                ...unapprovedTx,
-                txParams: {
-                  ...txParams,
-                  data: '',
-                },
-              },
-            },
-          })
```
Comment (onyb) [inline ui/app/ducks/metamask/metamask.js:330]: **Note to reviewer:**

I don't think all this stuff is relevant for Swap. Moreover, `editingTransactionId` is always unset in our case. Please let me know if this is incorrect.

File: ui/app/pages/swap/swap-content/swap-amount-row/swap-amount-row.component.js [hunk 2]
```diff
@@ -105,40 +104,22 @@ export default class SwapAmountRow extends Component {
...
-  render () {
-    const { gasTotal, inError } = this.props
-
-    return (
-      <SwapRowWrapper
-        label={`${this.context.t('amount')}:`}
-        showError={inError}
-        errorType="amount"
-      >
-        {gasTotal && <AmountMaxButton inError={inError} />}
```
Comment (onyb) [inline ui/app/pages/swap/swap-content/swap-amount-row/swap-amount-row.component.js:138]: **Note to reviewer:**

I'll probably add the "Max" feature back in a later PR.

File: ui/app/selectors/swap.js [hunk 3]
```diff
@@ -196,23 +192,6 @@ export function getSwapGasButtonGroupShown (state) {
   return state.swap.gasButtonGroupShown
 }
 
-export function getSwapTitleKey (state) {
```
Comment (onyb) [inline ui/app/selectors/swap.js:199]: **Note to reviewer:**

I think having "Swap Tokens" and "Swap ETH" is confusing. From a user's perspective, there should be no difference between the two, so I decided to set the header to just "Swap".

File: ui/app/pages/swap/swap-content/swap-amount-row/swap-amount-row.component.js [hunk 4]
```diff
@@ -105,40 +104,22 @@ export default class SwapAmountRow extends Component {
...
 
-  renderInput () {
-    const { amount, inError, swapFromToken } = this.props
+  render () {
+    const { amount, swapFromToken } = this.props
 
     return swapFromToken ?
       (
         <UserPreferencedTokenInput
-          error={inError}
```
Comment (onyb) [inline ui/app/pages/swap/swap-content/swap-amount-row/swap-amount-row.component.js:114]: **Note to reviewer:**

Removed it because the red border, in case of an error, was looking quite ugly. We still have the error message just below.

File: ui/app/pages/swap/swap-footer/swap-footer.component.js [hunk 5]
```diff
@@ -132,14 +87,31 @@ export default class SwapFooter extends Component {
...
+
   render () {
     return (
       <PageContainerFooter
-        onCancel={() => this.onCancel()}
         onSubmit={(e) => this.onSubmit(e)}
-        // disabled={this.formShouldBeDisabled()}
+        submitText="Review Swap"
+        disabled={this.reviewSwapButtonShouldBeDisabled()}
+        hideCancel
```
Comment (onyb) [inline ui/app/pages/swap/swap-footer/swap-footer.component.js:113]: **Note to reviewer:**

Cancel button is already in the header.

**Review by onyb — COMMENTED** [review]
Added some comments to help with reviews.

**Comment by Genysys** [comment]
Looks good!

A couple of comments

* Can we add tests to the components?
* I am not sure that we need the only mainnet flag as swaps should be possible  on Testnets , albeit for Test assets. 0x , and most of the DEXs are deployed on the testnets.

File: ui/app/ducks/metamask/metamask.js (hunk 1 above)
Comment (Douglashdaniel) [reply ui/app/ducks/metamask/metamask.js:330]: This is fine!

**Review by Douglashdaniel — CHANGES_REQUESTED** [review]
Hey welcome aboard! I Look forward to working with you! 😎
Looks good to me, just need to remove a console log.

File: ui/app/pages/swap/swap-content/swap-content.component.js [hunk 6]
```diff
@@ -47,13 +46,13 @@ export default class SwapContent extends Component {
...
+    getSwapQuotes().then((data) => {
+      this.setState({ quoteResult: data.quotes, quoteStatus: 'QUOTED' })
     })
   }
 
   fillOrder = () => {
-    const { quote , fillOrder } = this.props
+    const { quote, fillOrder } = this.props
     fillOrder().then((data) => {
       console.log('This Is The Quote', data)
```
[the remark this answers is not in the document]
Comment (onyb) [reply ui/app/pages/swap/swap-content/swap-content.component.js:57]: I'm actually going to clean up this whole file in the next PR. I'll take care of it there if that's okay.

**Comment by onyb** [comment]
Coming to the topic of local testing (already had some discussion with @Genysys on Slack), but I really think performing an end-to-end swap on a testnet / local ganache server is not possible without deploying the 0x contracts, even if you have the relevant custom token contracts deployed.

The way 0x Swap API works is that the response from their server is in fact the transaction parameters that must be used for creating the raw transaction.

```
GET https://api.0x.org/swap/v1/quote?some=thing
{
  "price": "...",
  "to": "0xdef1c0ded9bec7f1a1670819833240f027b25eff",   <--- 0x Swap Router on mainnet; won't work on testnet
  "data": "...",
  "value": "..."
  ...
}
```

What I did manage to test is the creation of a raw swap transaction with a mainnet account, but didn't sign/broadcast it of course.

**TLDR:** I think we should keep the Swap feature disabled on Mainnet, unless we come up with an easy way to completely replicate the entire swap ecosystem in the ganache server (0x router, token contracts, uniswap/sushiswap/kyber contracts, etc). Metamask does this too.

**Review by bbondy — APPROVED** [review]
